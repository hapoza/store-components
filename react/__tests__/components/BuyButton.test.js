import React from 'react'
import { render } from 'intl-helper'
import { fireEvent } from 'react-testing-library'

import BuyButton from '../../BuyButton'

describe('<BuyButton />', () => {
  const renderComponent = customProps => {
    const props = {
      ...customProps,
    }

    const comp = <BuyButton {...props}>Test</BuyButton>

    return render(comp)
  }

  function wait(ms) {
    var d = new Date()
    var d2 = null
    do {
      d2 = new Date()
    } while (d2 - d < ms)
  }

  it('should be rendered', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toBeDefined()
    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot normal', () => {
    const { asFragment } = renderComponent({ available: true, skuItems: [] })
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot unavailable', () => {
    const { asFragment } = renderComponent({ available: false, skuItems: [] })
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot loading', () => {
    const { asFragment } = renderComponent({ available: false })
    expect(asFragment()).toMatchSnapshot()
  })

  it('should call onAddStart', () => {
    const onAddStart = jest.fn()
    const { getByText } = renderComponent({
      available: true,
      skuItems: [],
      onAddStart,
    })

    fireEvent.click(getByText(/test/i))
    expect(onAddStart).toHaveBeenCalled()
  })

  it('should call onAddEnd', done => {
    const onAddFinish = jest.fn()
    const { getByText } = renderComponent({
      available: true,
      skuItems: [],
      onAddFinish,
    })

    fireEvent.click(getByText(/test/i))

    setTimeout(() => {
      expect(onAddFinish).toHaveBeenCalled()
      done()
    }, 300)
  })
})
