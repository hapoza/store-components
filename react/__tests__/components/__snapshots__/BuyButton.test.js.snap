// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuyButton /> should match snapshot loading 1`] = `
<DocumentFragment>
  <div
    class="content-loader-mock"
  />
</DocumentFragment>
`;

exports[`<BuyButton /> should match snapshot normal 1`] = `
<DocumentFragment>
  <button
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`<BuyButton /> should match snapshot unavailable 1`] = `
<DocumentFragment>
  <button
    disabled=""
    type="button"
  >
    <span>
      Unavailable
    </span>
  </button>
</DocumentFragment>
`;
